python/src/aiconfig/editor/server/server.py

```diff
@@ -194,15 +203,42 @@ async def run() -> FlaskResponse:
         params = request_json.get("params", aiconfig.get_parameters(prompt_name))  # type: ignore
         stream = request_json.get("stream", False)
         options = InferenceOptions(stream=stream)
-        run_output = await aiconfig.run(prompt_name, params, options)  # type: ignore
+
+        # Deepcopy the aiconfig prior to run so we can restore it in the case the run operation is cancelled or encounters some error
+        aiconfig_deep_copy = copy.deepcopy(aiconfig)
+
+        # Create a task to run the prompt, and associate the task with the cancellation token
+        # so that we can cancel the task if /cancel is invoked
+        task = asyncio.create_task(aiconfig.run(prompt_name, params, options))  # type: ignore
```

Comment:
I don't think this is going to work unless `run()` periodically yields control through an await, which it does not. I think we have to go heavy-handed and use threads.

Reply:
You're right :). It's been fixed with threads, thanks to @rossdanlm's changes for streaming